### PR TITLE
Adding CP features to ggplot visualization

### PR DESCRIPTION
**Pull request by gwaybio**
Also adding CP+DP features and the F1 score notebook

### CODE REVIEW

**Comment by gwaybio** [comment]
## Results summary

| Feature space | Number of top scoring phenotypes |
| :-------------: | :----------------------------------: |
| CP_and_DPtest | 10 |
| CPtest | 4 |
| DPtest | 2 |

image
 

## F1 Score results

f1_scores_multiclass

## PR Curve results

pr_curves_multiclass

**Comment by gwaybio** [comment]
@roshankern - I also had a question: what is `Weighted`? Should I include this class in the F1 score bar chart?

**Comment by roshankern** [comment]
> @roshankern - I also had a question: what is `Weighted`? Should I include this class in the F1 score bar chart?

`Weighted` refers to the weighted F1 score (the mean F1 score that takes into account occurrences of each label). I think this is good to include as it gives an idea of the overall performance of the model.

File: 7.figures/figure_themes.r [hunk 1]
```diff
@@ -18,6 +18,24 @@ shuffled_labels <- c(
...
 )
 
+feature_type_with_data_split_colors <- c(
+    "CP_and_DPtest" = "#C214CB",
+    "CP_and_DPtrain" = "#E88EED",
+    "CPtest" = "#EB4B4B",
+    "CPtrain" = "#F8B5B5",
+    "DPtest" = "#5158bb",
+    "DPtrain" = "#B5B9EA"
+)
```
Comment (roshankern) [inline 7.figures/figure_themes.r]: What made you choose these colors? I'm wondering if the contrast could be increased at all with a different color scheme.

File: 7.figures/nbconverted/f1_score_visualization.r [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        "shuffled" = "c",
+        "feature_type" = "c"
+    )
+) %>%
+    dplyr::select(!`...1`) %>%
+    dplyr::mutate(feature_type_with_data_split = paste0(feature_type, data_split)) %>%
+    dplyr::filter(
+        data_split == "test",
+        shuffled == "False"
+    )
```
Comment (roshankern) [inline 7.figures/nbconverted/f1_score_visualization.r:37]: Consider adding visualizations for F1 scores on training data and shuffled baseline on test data. I think the f1 scores on training data are particularly interesting to see how much different types of models overfit.

**Review by roshankern — APPROVED** [review]
I made a couple of small comments but overall LGTM!

**Comment by roshankern** [comment]
Another small comment: consider changing the final pr curve figure to 3 rows x 5 comments instead of 4x4 format. I think this would look better for a final figure, especially since there are 15 models. This is nitpicky, so feel free to ignore.

**Comment by roshankern** [comment]
There might be something to address in this PR now that the threshold values changed for multiclass models in #26.

File: 7.figures/figure_themes.r (hunk 1 above)
Comment (gwaybio) [reply 7.figures/figure_themes.r]: I picked purple (CP+DP) to indicate a combination of red (CP) and blue (DP), but I agree that i could choose colors with better contrast. I'll investigate

File: 7.figures/nbconverted/f1_score_visualization.r (hunk 2 above)
Comment (gwaybio) [reply 7.figures/nbconverted/f1_score_visualization.r:37]: My aim for the F1 score figure is to have a simple plot that includes a single number per phenotype, in fact, I might make it simpler!

The PR curves do the job of showing training vs. testing vs. shuffled baseline very well already :)

**Comment by gwaybio** [comment]
> Another small comment: consider changing the final pr curve figure to 3 rows x 5 comments instead of 4x4 format. I think this would look better for a final figure, especially since there are 15 models. This is nitpicky, so feel free to ignore.

This is a very good suggestion.

I've also implemented all other suggestions, I am going to merge!

**Comment by gwaybio** [comment]
> Weighted refers to the weighted F1 score (the mean F1 score that takes into account occurrences of each label). I think this is good to include as it gives an idea of the overall performance of the model.

One last thing - i changed this label to increase clarity
